omnibus/config/software/msodbcsql18.rb

```diff
@@ -0,0 +1,51 @@
+name "msodbcsql18"
+default_version "18.3.2.1-1"
+
+dependency "unixodbc"
+
+license "MICROSOFT SOFTWARE LICENSE"
+license_file "doc/msodbcsql18/LICENSE.txt"
+skip_transitive_dependency_licensing true
+
+# Dynamically build source url and sha256 based on the linux platform and architecture
+# The source url and sha256 are taken from the official Microsoft ODBC Driver for SQL Server page:
+if debian_target?
+  if arm_target?
+    # Debian ARM64 build
+    package_name = "#{name}_#{version}_arm64.deb"
+    source url: "https://packages.microsoft.com/debian/12/prod/pool/main/m/msodbcsql18/#{package_name}"
+    source sha256: "d9bb2d2e165e9d86f6b5de75b2baa24c9f0f25107471bcf82254d27f5dafff30"
+  else
+    # Debian AMD64 build
+    package_name = "#{name}_#{version}_amd64.deb"
+    source url: "https://packages.microsoft.com/debian/12/prod/pool/main/m/msodbcsql18/#{package_name}"
```

Comment:
nit: this could be shared for both arm & amd64

Reply:
(provided that the arch part is parameterized, since I missed the `amd64` in the package name)